src/Razor/test/Microsoft.CodeAnalysis.Razor.Workspaces.Test/ProjectSystem/ProjectStateTest.cs

```diff
@@ -667,82 +667,6 @@ public void ProjectState_WithHostProject_ResetsImportedDocuments()
         Assert.Equal(TestProjectData.SomeProjectFile1.FilePath, documentFilePath);
     }
 
-    [Fact]
-    public void ProjectState_WithProjectWorkspaceState_Removed()
-    {
-        // Arrange
-        var emptyProjectWorkspaceState = new ProjectWorkspaceState(ImmutableArray<TagHelperDescriptor>.Empty, csharpLanguageVersion: default);
-        var original = ProjectState.Create(Workspace.Services, _hostProject, emptyProjectWorkspaceState)
-            .WithAddedHostDocument(_documents[2], DocumentState.EmptyLoader)
-            .WithAddedHostDocument(_documents[1], DocumentState.EmptyLoader);
-
-        // Force init
-        var originalTagHelpers = original.TagHelpers;
-        var originalProjectWorkspaceStateVersion = original.ProjectWorkspaceStateVersion;
-
-        // Act
-        var state = original.WithProjectWorkspaceState(null);
-
-        // Assert
-        Assert.NotEqual(original.Version, state.Version);
-        Assert.Null(state.ProjectWorkspaceState);
-
-        var actualTagHelpers = state.TagHelpers;
-        var actualProjectWorkspaceStateVersion = state.ProjectWorkspaceStateVersion;
-
-        // The configuration didn't change, and the tag helpers didn't actually change
-        Assert.Same(original.ProjectEngine, state.ProjectEngine);
-
-        Assert.Equal(originalTagHelpers.Length, actualTagHelpers.Length);
-        for (var i = 0; i < originalTagHelpers.Length; i++)
-        {
-            Assert.Same(originalTagHelpers[i], actualTagHelpers[i]);
-        }
-
-        Assert.NotEqual(originalProjectWorkspaceStateVersion, actualProjectWorkspaceStateVersion);
-
-        Assert.NotSame(original.Documents[_documents[1].FilePath], state.Documents[_documents[1].FilePath]);
-        Assert.NotSame(original.Documents[_documents[2].FilePath], state.Documents[_documents[2].FilePath]);
-    }
-
-    [Fact]
-    public void ProjectState_WithProjectWorkspaceState_Added()
```

Comment:
This seems like a reasonable test. Is it not validating that when updated with the default state, it doens't clobber the tag helper info?

Reply:
The problem with this test as written is that I've changed `ProjectState` so that it's `ProjectWorkspaceState` is never null. Instead, it'll get a `ProjectWorkspaceState.Default` if a `ProjectWorkspaceState` isn't provided in `ProjectState.Create(...)`. The test is assuming `WithProjectWorkspaceState(ProjectWorkspaceState.Default)` will change a `ProjectState` that was created with a null `ProjectWorkspaceState`, but now it will do nothing. So, the asserts are all wrong.

I can change the asserts, but AFAICT, it would just be asserting that a newly created `ProjectState` with `ProjectWorkspaceState.Default` will be the same instance after `WithProjectWorkspaceState(ProjectWorkspaceState.Default)` is called. So, the test would also have a bad name. I figure that `ProjectState_WithProjectWorkspaceState_Changed` below covers the more important scenario. What do you think?

Reply:
Oops, I missed that the "before" state in the test was null, and I think I've misinterpretted what it was trying to do. Looking at the code in `ProjectState` again, I think removal here is fine.